lib/galaxy/datatypes/binary.py

```diff
@@ -169,6 +171,88 @@ def _get_samtools_version( self ):
                 break
         return version
 
+    def split(cls, input_datasets, subdir_generator_function, split_params):
+
+        # 1) by_rname -> splits the bam into files based on the chromosome
+        # 2) by_interval -> splits the bam into files based on  a defined bp length, and does so across the entire genome present in the BAM file
+        # 3) by_read -> splits the bam into files based on the number of reads encountered (if multiple files, all other files match the interval as the first)
+
+        if split_params is None:
+            return
+        if len(input_datasets) > 1:
+            raise Exception("BAM file splitting does not support multiple files")
+        input_file = input_datasets[0].file_name
+        ds = input_datasets[0]
+        input_file_bai = ""
+        for key, value in ds.metadata.items():
+            if isinstance( value, galaxy.model.MetadataFile ):
+                input_file_bai = value.file_name
+
+        # Link files with bam extensions in order to fulfil some tools requirements
+        commands = ["ln -sf " + input_file + " " + input_file + ".bam",
+                    "ln -sf " + input_file_bai + " " + input_file + ".bam.bai"]
```

Comment:
``input_file`` may have spaces in it - so this would need to be quoted. Also you shouldn't write into its current directory - it may leave files lingering in the object store directory that Galaxy has no record of. I think replacing these commands with something like:

 - Create a temp directory 
 - Link the bam file into that directory with ``os.symlink``.
 - Link bai in.

... would be a better approach.